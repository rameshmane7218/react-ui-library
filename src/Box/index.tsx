export { Box, BoxProps } from './Box';
