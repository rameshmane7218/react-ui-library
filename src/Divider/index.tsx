export { Divider, HrProps } from './Divider';
