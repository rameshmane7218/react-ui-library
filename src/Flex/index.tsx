export { Flex, FlexProps } from './Flex';
