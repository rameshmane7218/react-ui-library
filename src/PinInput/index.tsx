export { InputField, InputFieldProps } from './InputField';
export { PinInput, PinInputProps } from './PinInput';
export { PinInputExample } from './PinInputExample';
