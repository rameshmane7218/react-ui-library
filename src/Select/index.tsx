export { Select, SelectProps } from './Select';
export { Option, OptionProps } from './Option';
