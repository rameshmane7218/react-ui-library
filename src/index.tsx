export * from './Button';
export * from './Thing';
