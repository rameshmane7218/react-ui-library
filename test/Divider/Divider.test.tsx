import React from 'react';
import { render } from '@testing-library/react';
import { Divider } from '../../src';

describe('Divider Component', () => {
  it('should render the divider component without crashing', () => {
    render(<Divider />);
  });
  it('should render all variant', () => {
    const { asFragment } = render(
      <>
        <Divider variant="solid" />
        <Divider variant="dashed" />
        <Divider variant="dotted" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should have correct css properties', () => {
    const { getByTestId } = render(
      <Divider data-testid="divider" width={'10px'} />
    );
    expect(getByTestId('divider')).toHaveStyle({
      width: '10px',
      borderStyle: 'solid',
    });
  });
});
