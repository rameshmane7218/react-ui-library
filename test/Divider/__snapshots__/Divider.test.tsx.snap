// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render all variant 1`] = `
<DocumentFragment>
  <hr
    class="sc-jSUZER gfvkWP"
    width="100%"
  />
  <hr
    class="sc-jSUZER haKBbh"
    width="100%"
  />
  <hr
    class="sc-jSUZER hgbkZc"
    width="100%"
  />
</DocumentFragment>
`;
